persistence/src/main/java/org/cbioportal/genome_nexus/persistence/internal/EnsemblRepositoryImpl.java

```diff
@@ -109,4 +111,34 @@ public EnsemblTranscript findOneByHugoSymbolIgnoreCase(String hugoSymbol, String
         }
         return null;
     }
+
+    @Override
+    public HashMap<String, String> getHugoSymbolToEntrezGeneIdMap() {
+        HashMap<String, String> map = new HashMap<>();
+        List<EnsemblCanonical> transcripts = mongoTemplate.findAll(EnsemblCanonical.class, CANONICAL_TRANSCRIPTS_COLLECTION);
+        for (EnsemblCanonical transcript : transcripts) {
+            String[] previousSymbols = transcript.getPreviousSymbols();
+            String[] synonyms = transcript.getSynonyms();
+
+            if (!map.containsKey(transcript.getHugoSymbol())) {
+                map.put(transcript.getHugoSymbol(), transcript.getEntrezGeneId());
+            }
+            if (previousSymbols != null) {
+                for (String previousSymbol : previousSymbols) {
+                    if (!map.containsKey(previousSymbol)) {
```

Comment:
you don't really need to check if it contains the key, if it contains the key already it will override with the new (hopefully the same) value. This check adds additional complexity to the code, and we don't even handle the case when it contains different values for the same symbols.